ament_cmake_python/cmake/ament_python_install_package.cmake

```diff
@@ -42,34 +47,90 @@ function(_ament_cmake_python_install_package package_name)
     set(ARG_PACKAGE_DIR "${CMAKE_CURRENT_LIST_DIR}/${ARG_PACKAGE_DIR}")
   endif()
 
+  if(NOT ARG_VERSION)
+    # Use package.xml version
+    if(NOT _AMENT_PACKAGE_NAME)
+      ament_package_xml()
+    endif()
+    set(ARG_VERSION "${${PROJECT_NAME}_VERSION}")
+  endif()
+
   if(NOT EXISTS "${ARG_PACKAGE_DIR}/__init__.py")
     message(FATAL_ERROR "ament_python_install_package() the Python package "
       "folder '${ARG_PACKAGE_DIR}' doesn't contain an '__init__.py' file")
   endif()
 
-  _ament_cmake_python_register_environment_hook()
-
-  if(NOT PYTHON_INSTALL_DIR)
-    message(FATAL_ERROR "ament_python_install_package() variable "
-      "'PYTHON_INSTALL_DIR' must not be empty")
+  if(NOT ARG_SETUP_CFG)
+    if(EXISTS "${CMAKE_CURRENT_LIST_DIR}/setup.cfg")
+      set(ARG_SETUP_CFG "${CMAKE_CURRENT_LIST_DIR}/setup.cfg")
+    endif()
+  elseif(NOT IS_ABSOLUTE "${ARG_SETUP_CFG}")
+    set(ARG_SETUP_CFG "${CMAKE_CURRENT_LIST_DIR}/${ARG_SETUP_CFG}")
   endif()
-  install(
-    DIRECTORY "${ARG_PACKAGE_DIR}/"
-    DESTINATION "${PYTHON_INSTALL_DIR}/${package_name}"
-    PATTERN "*.pyc" EXCLUDE
-    PATTERN "__pycache__" EXCLUDE
+
+  set(build_dir "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_python/${package_name}")
+  file(RELATIVE_PATH source_dir "${build_dir}" "${ARG_PACKAGE_DIR}")
+
+  string(CONFIGURE "\
+import os
+from setuptools import find_packages
+from setuptools import setup
+
+setup(
+  name='${package_name}',
+  version='${ARG_VERSION}',
+  packages=find_packages(
+    where=os.path.normpath('${source_dir}/..'),
+    include=('${package_name}', '${package_name}.*')),
+  package_dir={'${package_name}': '${source_dir}'},
+  package_data={'': ['*.*']}
+)
+" setup_py_content)
+
+  file(GENERATE
+    OUTPUT "${build_dir}/setup.py"
+    CONTENT "${setup_py_content}"
   )
-  if(NOT ARG_SKIP_COMPILE)
-    # compile Python files
-    install(CODE
-      "execute_process(
-        COMMAND
-        \"${PYTHON_EXECUTABLE}\" \"-m\" \"compileall\"
-        \"${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}/${package_name}\"
-      )"
+
+  if(ARG_SETUP_CFG)
+    add_custom_command(
+      OUTPUT "${build_dir}/setup.cfg"
+      COMMAND ${CMAKE_COMMAND} -E copy ${ARG_SETUP_CFG} ${build_dir}/setup.cfg
```

Comment:
Might need to quote the arguments here if the file names contain spaces.

Reply:
Good catch! See https://github.com/ament/ament_cmake/pull/316/commits/7d8cc5c599c10cabe36651fbe082b11995c48f43.